### PR TITLE
DROID-2756 Editor | Fix | Corrected text ellipsizing logic in custom TextInputWidget

### PR DESCRIPTION
**Pull request by Nadezhda-Gurova**
---

- [ ] I understand that contributing to this repository will require me to agree with the [CLA](https://github.com/anyproto/.github/blob/main/docs/CLA.md)

---

### What type of PR is this? (check all applicable)

- [ ] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI

### Added tests?

- [ ] 👍 yes
- [ ] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help

### Added to documentation?

- [ ] 📜 README.md
- [ ] 📓 [tech-docs](https://github.com/anyproto/tech-docs)
- [ ] 🙅 no documentation needed

### CODE REVIEW

**Comment by Nadezhda-Gurova** [comment]
Before fix
 
Screenshot 2024-08-20 at 12 07 02

After fix
Screenshot_20240829_144715

File: presentation/src/test/java/com/anytypeio/anytype/presentation/history/VersionHistoryViewModelTest.kt [hunk 1]
```diff
@@ -314,7 +315,7 @@ class VersionHistoryViewModelTest {
                 groups = buildList {
                     add(
                         VersionHistoryGroup(
-                            id = versionsNew[8].id,
+                            id = versionsNew[0].id,
```
Comment (uburoiubu) [inline presentation/src/test/java/com/anytypeio/anytype/presentation/history/VersionHistoryViewModelTest.kt:318]: please revert these changes, this test should be fixed in the "main" branch

File: core-ui/src/main/java/com/anytypeio/anytype/core_ui/features/editor/holders/media/File.kt [hunk 2]
```diff
@@ -42,40 +44,58 @@ class File(val binding: ItemBlockFileBinding) : Media(binding.root), Decoratable
...
+                Spannable.SPAN_EXCLUSIVE_EXCLUSIVE
             )
-            name.setText(spannable, TextView.BufferType.SPANNABLE)
+            name.text = spannable
         } else {
-            name.setText(item.name, TextView.BufferType.SPANNABLE)
+            name.text = item.name
         }
 
         applySearchHighlight(item)
```
Comment (uburoiubu) [inline core-ui/src/main/java/com/anytypeio/anytype/core_ui/features/editor/holders/media/File.kt:101]: consider testing the following corner-case scenario: when searching inside the object (go to three-dots menu, then select "search", then enter search query containing characters from the original filename), what happens if the search highlight is applied to the ellipsized text sequence? there might be an issue with the span ranges here, because these text ranges are applied to the original filename.

**Review by uburoiubu — CHANGES_REQUESTED** [review]

**Comment by uburoiubu** [comment]
consider adding a screenshot with the very long filename — to test this corner-case scenario.
